ably/rest_client.go

```diff
@@ -35,6 +36,23 @@ func query(fn func(context.Context, string, interface{}) (*http.Response, error)
 	}
 }
 
+type Raw []byte
+
+func (r *Raw) UnmarshalJSON(data []byte) error {
+	*r = data
+	return nil
+}
+
+func (r *Raw) CodecEncodeSelf(*codec.Encoder) {
+	panic("Raw cannot be used as encoder")
```

Comment:
I'm not sure the reason to use panic here. Panic will kill execution and cause an exit with a non-zero status, which feels a bit severe. Assuming this method is needed to satisfy an interface, is there a better way to handle the scenario when `CodecEncodeSelf` is called?

Reply:
This function will never be called. This is the same panic line taken from `rest_channel.go`

Reply:
I don't like the `panic` there either, but at least we are being consistent :joy: 

Analysing our use of `panic` in this SDK is probably a bigger task for another day.